artiq/gateware/eem.py

```diff
@@ -502,12 +502,12 @@ def io(eem, eem_aux, iostandard):
 
     @classmethod
     def add_std(cls, target, eem, eem_aux=None, eem_aux2=None, ttl_out_cls=None,
-            iostandard=default_iostandard):
+            iostandard=default_iostandard, roi_engine_count=16):
```

Comment:
Since the json schema already include 16 in the default, no need to include 16 here again.

Reply:
Would it be ok to break backwards compatibility for that? In principle, there could be other users of `artiq.gateware.eem.Grabber.add_std` than through the Kasli target builder.

Side question: would you be happy with having `roi_engine_count` as keyword-only argument?

```suggestion
            iostandard=default_iostandard, *, roi_engine_count):
```

Reply:
The `add_std` is not a user facing api so it's okay.
And no, as other peripherals' `add_std` don't use keyword-only argument.

Reply:
If the change is backwards compatible, would you consider backporting the change to `release-8`?

(The PHY gateware component also has 16 as default value for its `roi_engine_count` argument. If the default should be deduplicated, that one should be removed too.)

Reply:
If I remove the default value and don't use a keyword-only argument, the `roi_engine_count` argument needs to be latest after `eem`, which will be a much bigger and less intuitive change. Which of the two variants would you prefer?

```python
def add_std(self, target, roi_engine_count, eem, eem_aux=None, ...):
    ...
```

or

```python
def add_std(self, target, eem, roi_engine_count, eem_aux=None, ...):
    ...
```

> The general guideline is to make the code as good as possible without caring about backward compatibility on the beta branch

What about backports? This patch would be really nice to have on `release-8`. It isn't new functionality, since it only exposes a setting that was already there.

The keyword-only argument would be a better option than a compulsory positional-or-keyword argument, since it reduces errors at the call site, especially with that many arguments.